compiler/compiler.go

```diff
@@ -421,6 +437,16 @@ func (c *compileCtx) getBoxedType(t protobuf.Type) protobuf.Type {
 	}
 }
 
+func (c *compileCtx) getArrayType(name string) protobuf.Type {
```

Comment:
This appears to return the same type no matter what. Is this switch even needed?

Reply:
Nope, it isn't